src/coprocessor/dag/rpn_expr/impl_arithmetic.rs

```diff
@@ -0,0 +1,216 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use cop_codegen::RpnFunction;
+
+use super::types::RpnFnCallPayload;
+use crate::coprocessor::codec::data_type::*;
+use crate::coprocessor::codec::{self, Error};
+use crate::coprocessor::dag::expr::EvalContext;
+use crate::coprocessor::Result;
+use std::fmt::Debug;
+
+#[derive(Debug, RpnFunction)]
+#[rpn_function(args = 2)]
+pub struct RpnFnArithmetic<A: ArithmeticOp> {
+    _phantom: std::marker::PhantomData<A>,
+}
+
+impl<A: ArithmeticOp> RpnFnArithmetic<A> {
+    pub fn new() -> Self {
+        Self {
+            _phantom: std::marker::PhantomData,
+        }
+    }
+
+    #[inline]
+    fn call(
+        _ctx: &mut EvalContext,
+        _payload: RpnFnCallPayload<'_>,
+        arg0: &Option<A::T>,
+        arg1: &Option<A::T>,
+    ) -> Result<Option<A::T>> {
+        if let (Some(lhs), Some(rhs)) = (arg0, arg1) {
+            A::calc(lhs, rhs)
+        } else {
+            // All arithmetical functions with a NULL argument return NULL
+            Ok(None)
+        }
+    }
+}
+
+impl<A: ArithmeticOp> Clone for RpnFnArithmetic<A> {
+    fn clone(&self) -> Self {
+        Self::new()
+    }
+}
+
+impl<A: ArithmeticOp> Copy for RpnFnArithmetic<A> {}
```

Comment:
Why not derive `Copy` directly?

Reply:
A rust bug https://github.com/rust-lang/rust/issues/26925 ...